RFCs/first-person-mercator-viewport-rfc.md

```diff
@@ -0,0 +1,230 @@
+
+Instead it becomes interesting to give the user freedom to move the camera around in the 3D scene, as if being present inside it. It might for example be desirable to let the user take the viewpoint of a pedestrian on the street or take the driver’s seat of a car, and to enable the user to look around or even up (away from the ground/map), as if in a first person 3D game.
+
+
+## Problem Description
+
+There are multiple issues with the current camera system
+
+* **Cartographic projection** deck.gl's current viewport system treats Web Mercator viewports as a special subclass, separate from normal perspective and orthographic cameras. This means that the `PerspectiveViewport` which would seem the natural choice for first person camera cannot be used with geospatial coordinates.
+
+* **Camera Positioning** - When working with the traditional deck.gl `WebMercatorViewport`, the focus is not on specifying the position of the camera, but rather what position (lng/lat) we are looking at, and at what zoom, pitch and bearing we are looking at it. An appropriate camera position is then implicitly calculated (in a way that perfectly matches the way the underlying mapbox camera system), however the application does not have access to the camera position.
+
+* **Camera Coordinates** - Our hybrid applications usually use data with coordinates in `METER_OFFSET`s, transformed by modelMatrices. Being able to position camera(s) in the same ways (relative coordinates + matrices) is critical as this would dramatically simplify applications by allowing them to set camera positions inside existing objects simply by copying their coordinates.
+
+* **FOV sync** - Since our 3D geometry is completely general, we have the ability to show it from any camera position, height, and direction (including dual cameras for stereoscopic view etc).
+However, to synchronize our 3D data with external perspective-enabled map systems like Mapbox GL, we need to adjust our 3D camera to follow the map’s when in “hybrid” map/3D mode.
+The map system typically locks in FoV (Field of view), viewing angle (always pitched downward) and altitude (camera height over the “ground”).
+When rendering 3D environments on top of pre-rendered video (e.g. overlaying perception data on top of vehicle log cameras)
+
+
+## Proposal: Support for Cartographic Projection in all Viewports
+
+In deck.gl 4.1 only the WebMercatorViewport can handle layers with geospatial coordinates, as it is the only viewport that can produce the uniforms required by the project module.
+
+By moving some of the required properties into the base viewport, it would be possible to extend cartographic projection to all cameras.
+
+Note: There are subtleties around the positioning of the camera which is handled in the next proposal.
+
+
+### Viewport
+
+The current base viewport class would be extended with a lnglat property
+
+- lnglat anchor
```

Comment:
> lnglat anchor is this camera position or camera center?

The camera position is the lng lat anchor + any meter offset specified by position. Just like any object is positioned in a METER_OFFSETS layer. I will clarify.